pkg/collector/distinct_string_collector.go

```diff
@@ -7,51 +7,73 @@ import (
 )
 
 type DistinctString struct {
-	values   map[string]struct{}
-	new      map[string]struct{}
-	maxLen   int
-	currLen  int
-	totalLen int
-	mtx      sync.Mutex
+	values      map[string]struct{}
+	new         map[string]struct{}
+	maxLen      int
+	currLen     int
+	diffEnabled bool
+	limExceeded bool
+	mtx         sync.Mutex
 }
 
 // NewDistinctString with the given maximum data size. This is calculated
 // as the total length of the recorded strings. For ease of use, maximum=0
 // is interpreted as unlimited.
 func NewDistinctString(maxDataSize int) *DistinctString {
 	return &DistinctString{
-		values: make(map[string]struct{}),
-		new:    make(map[string]struct{}),
-		maxLen: maxDataSize,
+		values:      make(map[string]struct{}),
+		new:         make(map[string]struct{}),
```

Comment:
Nit: no need to allocate this in no-diff version

Reply:
removed